content/Hardware Support/UNO/Whats-the-difference-between-UNO-R3-and-UNO-R4-boards.md

```diff
@@ -3,56 +3,91 @@ title: What's the difference between UNO R3 and UNO R4 boards?
+### Selected differences
 
-To use the new UNO R4 boards, install the new **Arduino UNO R4 Boards** package.
+#### Different behavior on runtime errors
 
----
+If the board encounters a runtime error while running an uploaded sketch, you need to double-press the reset button to put it into bootloader mode before uploading a new sketch.
 
-## Does the UNO R4 replace the UNO R3?
+Note that unlike some Mbed OS boards, runtime errors do not result in a flashing LED.
 
-No, the Arduino UNO R3 will still be available and supported for makers who want to work with its 8-bit AVR microcontroller.
+The UNO R4 WiFi board features an error-catching mechanism that detects operations that caused a runtime crash, as opposed to the usual silent behavior of a crashed board.
 
----
+#### Different board package
 
-## Compatibility
+Previous UNO boards (except the UNO WiFi Rev2) were AVR-based, and used the **Arduino AVR Boards** package. For UNO R4, install the **Arduino UNO R4 Boards** package.
+
+## Compatibility {#compatibility}
 
 ### Can I use hardware compatible with the UNO R3 with an UNO R4 board?
 
 UNO R4 boards maintain the same mechanical and electrical compatibility, allowing you to seamlessly use your existing shields and hardware with the new board.
 
-### Can I use my sketch developed for UNO R3 with an UNO R4 board?
+> [!NOTE]
+> Third-party UNO shields are compatible with R4 in terms of voltage and pinout. However,  software compatibility might require some porting work, so it should be checked with shield manufacturers before purchase.
 
-Yes, if your sketch was developed using the [Arduino API](https://docs.arduino.cc/language-reference/).
+### Can I use my sketch developed for UNO R3 with an UNO R4 board?
 
-If your sketch uses instructions specific to the AVR architecture, they will have to be changed to ensure compatibility.
+Yes, if your sketch was developed using the [Arduino API](https://docs.arduino.cc/language-reference/). If your sketch uses instructions specific to the AVR architecture, they will have to be changed to ensure compatibility.
 
 ### Are libraries for UNO R3 also compatible with the UNO R4 boards?
 
-Libraries that are based on the [Arduino API](https://docs.arduino.cc/language-reference/) will work as is.
-
-Libraries that use AVR-specific instructions are not compatible with the architecture of UNO R4 boards. However, there are libraries that have already been ported as part of our early adopters program.
-
-<!-- markdownlint-disable-file HC001 -->
+- Libraries that are based on the [Arduino API](https://docs.arduino.cc/language-reference/) will work as is.
+
+- Libraries that use AVR-specific instructions are not compatible with the architecture of UNO R4 boards. However, some of these libraries have already been ported. For more details, refer to the [UNO R4 Library Compatibility repository](https://github.com/arduino/uno-r4-library-compatibility).
```

Comment:
I feel the heading structure is a bit complex here. How about turning "Compatability" into a FAQ section? It may be better to have the FAQs after the table, so more FAQs can be added without pushing the table to far down. For the information in "Selected differences", cover it in FAQs or leave it out.

Reply:
[8dcf340](https://github.com/arduino/help-center-content/pull/459/commits/8dcf340a775bc5ef4926b18946531cf1b248c47f)